### DISCUSSION TITLE
differential reduce to pole

### DISCUSSION THREAD

**Discussion by prince-Ha** [fatiando/community#93] [Q&A] [answered]
Hello 
I'm asking about transformation
When can I use reduce to pole and differential reduce to pole(drtp) in fatiando? I think it'd be nice to have that program

Have a nice day.

**Comment by santisoler** [comment, marked as answer]
Hi @prince-Ha. The `harmonica.reduction_to_pole` function is already in our `main` branch, you can see its documentation on the `dev` version of Harmonica docs (https://www.fatiando.org/harmonica/dev/api/generated/harmonica.reduction_to_pole.html#harmonica.reduction_to_pole). If you install the development version of Harmonica you can try it out right now. Otherwise, I'm planning to make the v0.6.0 release by the end of this month or early March 🤞🏼

**Reply by LL-Geo** [reply]
Hi @prince-Ha, as @santisoler said that the RTP function is in harmonica right now. If your study area is less than 2°, RTP should be fairly safe.
If you want to use differential reduce to pole(drtp), one option is to divide the data into overlying blocks. You can apply RTP function in each block and then combine these overlapping blocks together. 
The other option is the method by [Cooper and Cowan](https://www.sciencedirect.com/science/article/pii/S0098300405000634?via%3Dihub). Yeah, please open an Issue in Harmonica. And we can decide which one is the better way for this purpose. 😄

**Reply by prince-Ha** [reply]
@santisoler  I'm relying on fatiando a lot. Thank you. I'll apply rtp after updating it. 
@LL-Geo I want to use [Cooper and Cowan](https://www.sciencedirect.com/science/article/pii/S0098300405000634?via%3Dihub) but I can't access the code homepage.


- Does it mean that we should combine the data by applying different inclination and declination angles for each 2 degrees?
ok I will apply it on my data

 
- How does this data determine inclination and declination? Is it okay to apply it at any specific time because there is little change in the inclination and declination.

- data link : https://www.ncei.noaa.gov/access/metadata/landing-page/bin/iso?id=gov.noaa.ngdc.mgg.geophysical_models:EMAG2_V3

- Is rtp already applied to this data?
thank you

**Reply by LL-Geo** [reply]
Yea..It seems that the URL has been replaced...

- _Does it mean that we should combine the data by applying different inclination and declination angles for each 2 degrees?_

It depends on your focus area. I guess you are going to use EMAG2. How large data do you need? The whole globe, a continent, or a country? Ideally, you can apply RTP in each 1-degree data, with about 10-15% data overlap.

- _How does this data determine inclination and declination? Is it okay to apply it at any specific time because there is little change in the inclination and declination._

You can use the mean value of each 1-degree block. If you have the time for the data collection, then use the time when data is collected. If not, then you could probably choose the date when most data is collected or the middle of that project time...

- _Is rtp already applied to this data?_

I think EMAG2 is just TMI value, but better check the document and the associated paper.

PS: It looks like [Arkani-Hamed, 2007](https://library.seg.org/doi/abs/10.1190/1.2399370?journalCode=gpysa7) would fit your purpose. The space domain method might cause some distortion for long wavelength information.

**Reply by LL-Geo** [reply]
Btw... I emailed the IAMG office. They kindly provide the code attached to the paper [Cooper and Cowan](https://www.sciencedirect.com/science/article/pii/S0098300405000634?via%3Dihub). If you use Matlab, you could try this. 
[v31-08-05.zip](https://github.com/fatiando/community/files/10840135/v31-08-05.zip)

**Reply by prince-Ha** [reply]
@LL-Geo I apologize for the delay in my response due to suddenly having too many things to attend to. I did try to apply DRTP using the code from the link https://docs.generic-mapping-tools.org/6.4/supplements/potential/grdredpol.html?highlight=grdredpol, but I think I need to double-check if I processed the data correctly. I will compare the two datasets using Cooper's method that you provided. Thank you so much for your attention!

**Comment by santisoler** [comment]
Regarding the DRTP, are you talking about the method introduced in [this article by Arkani-Hamed](https://library.seg.org/doi/abs/10.1190/1.2399370?journalCode=gpysa7)?

It might sound like a nice idea! Would you like to open an Issue in Harmonica describing it a little bit, adding links to the article and explaining the main advantages over the classic FFT reduction to the pole transformation?

**Reply by prince-Ha** [reply]
@santisoler   I'll read the papers and post the issue. 

thank yout : )
